dev/provider_packages/prepare_provider_packages.py

```diff
@@ -747,15 +750,31 @@ def convert_git_changes_to_table(changes: str, base_url: str) -> str:
         if line == "":
             continue
         full_hash, short_hash, date, message = line.split(" ", maxsplit=3)
-        table_data.append((f"[{short_hash}]({base_url}{full_hash})", date, message))
-    return tabulate(table_data, headers=headers, tablefmt="pipe")
+        message_without_backticks = message.replace("`", "'")
+        table_data.append(
+            (
+                f"[{short_hash}]({base_url}{full_hash})"
+                if markdown
+                else f"`{short_hash} <{base_url}{full_hash}>`_",
+                date,
+                f"`{message_without_backticks}`" if markdown else f"``{message_without_backticks}``",
+            )
+        )
+    table = tabulate(table_data, headers=headers, tablefmt="pipe" if markdown else "rst")
+    header = ""
+    if not markdown:
+        header += f"\n\n{print_version}\n" + "." * len(print_version) + "\n\n"
+        release_date = table_data[0][1]
+        header += f"Latest change: {release_date}\n\n"
+    return header + table
```

Comment:
Have you considered using rich for rendering the table instead of using tabulate + custom headers? I think using rich table make take less code and be easier to understand

Reply:
This is a table that lands in commit.rst and I think it's really eeasy to follow (those are just lists of commits). They are not supposed to be updated/modified etc - they are always automatically generted  I do not think we can make them more readable to be honest :). 

See https://user-images.githubusercontent.com/595491/106393635-96531a80-63f8-11eb-8fb8-f2135f8054d8.png :)